compiler/optimizer/Operation.cpp

```diff
@@ -0,0 +1,347 @@
+         result.type = Double;
+         result.data.dconst = (double_t)operand1.data.lconst;
+         break;
+      }
+      case TR::l2b:
+      {
+         operand1 = nodeValueMap[node->getChild(0)->getGlobalIndex()];
+         result.type = Int8;
+         result.data.bconst = (int8_t)operand1.data.lconst;
+         break;
+      }
+      case TR::l2s:
+      {
+         operand1 = nodeValueMap[node->getChild(0)->getGlobalIndex()];
+         result.type = Int16;
+         result.data.sconst = (int16_t)operand1.data.lconst;
+         break;
+      }
+      case TR::l2a:
+      {
+         operand1 = nodeValueMap[node->getChild(0)->getGlobalIndex()];
+         result.type = Address;
+         result.data.aconst = (uintptrj_t)operand1.data.lconst;
+         break;
+      }
+      case TR::lcmpeq:
+      {
+         operand1 = nodeValueMap[node->getChild(0)->getGlobalIndex()];
+         operand2 = nodeValueMap[node->getChild(1)->getGlobalIndex()];
+         result.type = Boolean;
+         result.data.lconst = operand1.data.lconst == operand2.data.lconst ? true : false;
+         break;
+      }
+      case TR::lcmpne:
+      {
+         operand1 = nodeValueMap[node->getChild(0)->getGlobalIndex()];
+         operand2 = nodeValueMap[node->getChild(1)->getGlobalIndex()];
+         result.type = Boolean;
+         result.data.lconst = operand1.data.lconst != operand2.data.lconst ? true : false;
+         break;
+      }
+      case TR::lcmplt:
+      {
+         operand1 = nodeValueMap[node->getChild(0)->getGlobalIndex()];
+         operand2 = nodeValueMap[node->getChild(1)->getGlobalIndex()];
+         result.type = Boolean;
+         result.data.lconst = operand1.data.lconst <  operand2.data.lconst ? true : false;
+         break;
+      }
+      case TR::lcmpge:
+      {
+         operand1 = nodeValueMap[node->getChild(0)->getGlobalIndex()];
+         operand2 = nodeValueMap[node->getChild(1)->getGlobalIndex()];
+         result.type = Boolean;
+         result.data.lconst = operand1.data.lconst >= operand2.data.lconst ? true : false;
+         break;
+      }
+      case TR::lcmpgt:
+      {
+         operand1 = nodeValueMap[node->getChild(0)->getGlobalIndex()];
+         operand2 = nodeValueMap[node->getChild(1)->getGlobalIndex()];
+         result.type = Boolean;
+         result.data.lconst = operand1.data.lconst >  operand2.data.lconst ? true : false;
+         break;
+      }
+      case TR::lcmple:
+      {
+         operand1 = nodeValueMap[node->getChild(0)->getGlobalIndex()];
+         operand2 = nodeValueMap[node->getChild(1)->getGlobalIndex()];
+         result.type = Boolean;
+         result.data.lconst = operand1.data.lconst <= operand2.data.lconst ? true : false;
+         break;
+      }
+      case TR::lcmp:
+      {
+         operand1 = nodeValueMap[node->getChild(0)->getGlobalIndex()];
+         operand2 = nodeValueMap[node->getChild(1)->getGlobalIndex()];
+         result.type = Int8;
```

Comment:
I thought maybe we didn't need so much memory for such a small range, but I believe you're right. Fixed.

Reply:
Actually, using `Int8` would have been fine if you had changed the result to be in `.data.bconst`. Either way, what matters is that the two should match.

Reply:
Also, because you're using a union for the data, the amount of memory used will end up being the same no matter what type you say you're using. :wink: